src/engine/strat_engine/engine.rs

```diff
@@ -113,12 +118,29 @@ impl Engine for StratEngine {
         let pool = StratPool::initialize(name, &dm, blockdev_paths, redundancy, force)?;
 
         let uuid = pool.uuid();
+        devlinks::pool_added(pool.name());
         self.pools.insert(pool);
+
         Ok(uuid)
     }
 
     fn destroy_pool(&mut self, uuid: PoolUuid) -> EngineResult<bool> {
-        destroy_pool!{self; uuid}
+        if let Some(ref pool) = self.pools.get_by_uuid(uuid) {
+            if pool.has_filesystems() {
+                return Err(EngineError::Engine(ErrorEnum::Busy,
+                                               "filesystems remaining on pool".into()));
+            };
+        } else {
+            return Ok(false);
+        }
+
+        let pool = self.pools
+            .remove_by_uuid(uuid)
+            .expect("Must succeed since $s.pool.get_by_uuid() returned a value");
+        let pool_name = pool.name().to_owned();
+        pool.destroy()?;
```

Comment:
Would it be better to remove the the devlink before you remove the actual pool?  Trying to think which is better, leaving a symlink for a pool that failed to be removed or missing a symlink for a pool that still exists?

I keep thinking it should be done in reverse order of creation.